src/system.rs

```diff
@@ -214,6 +274,31 @@ impl TypeStore {
         type_id
     }
 
+    /// Adds the [NamedTypeDefinition] and the associated name as the imports of [TypeStore]
+    ///  and returns the [TypeId] for it. If the name already exists in the [TypeStore] it returns the associated [TypeId]
+    pub fn add_import_type(&mut self, type_def: TypeDefinitionImpl) -> TypeId {
+        let name = type_def.name().as_ref().unwrap();
+        if let Some(exists) = self.imports.get(name) {
+            return exists.to_owned();
+        }
+        let type_id = self.types_by_id.len();
+        self.imports.insert(name.to_owned(), type_id);
+        self.types_by_id.push(TypeDefinition::Named(type_def));
+        type_id
+    }
+
+    /// Adds the [NamedTypeDefinition] and the associated alias as the [TypeAlias] import in [TypeStore]
+    ///  and returns the [TypeId] for it. If the name already exists in the [TypeStore] it returns the associated [TypeId]
+    pub fn add_alias_import_type(&mut self, alias: &str, type_def: TypeDefinitionImpl) -> TypeId {
```

Comment:
How about:
```suggestion
    pub fn add_aliased_import_type(&mut self, alias: &str, type_def: TypeDefinitionImpl) -> TypeId {
```
or, to be consistent with the grammar's element naming:
```suggestion
    pub fn add_import_type_alias(&mut self, alias: &str, type_def: TypeDefinitionImpl) -> TypeId {
```